refactor: [M3-9645] - Remove `flags` as a prop and move `useIsGeckoEnabled` out of `RegionSelect`

pmakode-akamai opened the pull request:
## Description 📝

- Follow-up to #11790 

~`FlagSet` and feature flag-related files are likely to be moved to `@linode/shared`, so we don't need to pass them as props to `RegionSelect`~

Since we haven't decided on how to handle `flags` yet, I've taken a slightly different approach for now:
- Instead of passing flags to `RegionSelect` & `RegionMultiSelect`, we are now passing `isGeckoLAEnabled`
- Moved the `useIsGeckoEnabled` hook out of `RegionSelect` (since `useIsGeckoEnabled` hook is dependent on Gecko feature flags(`enabled` & `la`) and this hook is a dependency of `RegionSelect`, too)
  - Currently, kept in `@linode/shared` since this hook depends on the regions query. I think this could be further improved by passing the regions query data to the hook as a prop, which would allow us to move it to `@linode/utilities`, but for now, I'm leaving it as is.
- A bit of reorganization in the shared package: Moved components like `LinodeSelect` to the `/components` directory and created a new `/hooks` directory to keep the hooks that depend on queries

## Changes  🔄
- Revert feature `flags` as a prop to `RegionSelect`
- Move `useIsGeckoEnabled` out of `RegionSelect`
- Passing `isGeckoLAEnabled` as a prop to `RegionSelect` & `RegionMultiSelect`

## Target release date 🗓️
N/A

## How to test 🧪
- Confirm all checks pass
- Confirm all the necessary imports are updated

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support
</details>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

hkhalil-akamai commented:
> `FlagSet` and feature flag-related files are likely to be moved to `@linode/shared`

I'm not sure if this has been decided yet. The question is if we want MFEs to share feature flags with us or handle it themselves. @abailly-akamai?

pmakode-akamai commented:
@abailly-akamai @hkhalil-akamai @jaalah-akamai 

I also believe passing `flags` as a prop to `RegionSelect` is not ideal in this case. Even if we do decide to pass them, we would need to import the `FlagSet` type in `RegionSelect` (which we are currently doing), which could cause the abstraction to lose its meaning and clarity.

Here are a couple of options I could think of:
1. If we decide MFEs to share feature flags with us (likely through a shared package like `@linode/shared` or `@linode/flags`):
    - In this case, the current implementation in the PR would make sense and align with this approach once flags-related files are migrated to that package.
    - Pros:
      - Centralized
      - Consistent flag usage
    - Cons:
       - requires an extra package / need to keep flags-related files in other package  (likely `@linode/shared`)
      
3. If we decide MFEs should handle their own feature flags:
    - In this case, it may be ideal to pass `Gecko flags` (booleans) directly to `RegionSelect` instead of `flags` with `FlagSet` type since `Gecko` feature flags are dependent on `RegionSelect`. This would allow us to avoid importing additional types (like these [L-15](https://github.com/linode/manager/pull/11918/files#diff-b4fe8f4000b39252c9e82fa5cb5a4140b91f54efb21a5ad4b6f8e4ec7e16fa30L15), [L-10](https://github.com/linode/manager/pull/11918/files#diff-4c58cabedde0fcc430681cead5a1ae109339e366e970192eb813cf3a65bbb920L10), etc) and simplify the implementation.
    - Pros:
       - More Flexibility for each MFE
    - Cons: 
        - More feature flags maintenance could require if a team manages multiple MFEs
        - Inconsistent across MFEs
        - Potential code duplication
        
Let me know your thoughts!

abailly-akamai commented:
@pmakode-akamai @hkhalil-akamai also including @jaalah-akamai 

- I like the idea of `isGeckoLAEnabled` to be a boolean on the select and not have to pass flags at all. This can defer our decision on how we handle flags.
- I do think MFEs _**should**_ have access to our flag as needed (optional) but perhaps this isn't something we need to answer right away?

pmakode-akamai commented:
@abailly-akamai @hkhalil-akamai 

Since we haven't decided on how to handle flags yet, I've taken a slightly different approach for now:
- Instead of passing flags to `RegionSelect`, we are now passing `isGeckoLAEnabled`
- Moved the `useIsGeckoEnabled` hook out of `RegionSelect` (since `useIsGeckoEnabled` hook is dependent on Gecko feature flags(`enabled` & `la`) and this hook is a dependency of `RegionSelect`, too)
    - Currently, kept in `@linode/shared` since this hook depends on the regions query. I think this could be further improved by passing the regions query data to the hook as a prop, which would allow us to move it to `@linode/utilities`, but for now, I'm leaving it as is.
- I did a bit of reorganization in the shared package: I moved components like `LinodeSelect` to the `/components` directory and created a new `/hooks` directory to keep the hooks that depend on queries

Let me know if these changes look good to you!

hkhalil-akamai commented:
This iteration makes sense to me. Maybe we should slightly update [shared/README.md](packages/shared/README.md) which currently only mentions that the package contains components.

abailly-akamai reviewed (approved):
Thanks you! this is a good approach to handling this - appreciate your patience while finding the best pattern for this component 👍

hkhalil-akamai reviewed (approved):
Echoing what @abailly-akamai said. Thanks for being flexible!

✅ Tested `RegionSelect`s across the app
✅ Code review looks good

pmakode-akamai commented:
Merging - The test failure in `timerange-verification.spec.ts` is unrelated and passes on my local
